tests/test_actions.py

```diff
@@ -0,0 +1,428 @@
+
+import unittest
+
+import astropy.units as u
+import numpy as np
+import pandas as pd
+from lsst.analysis.tools.actions.scalar.scalarActions import (
+    ApproxFloor,
+    CountAction,
+    MeanAction,
+    MedianAction,
+    SigmaMadAction,
+    StdevAction,
+)
+from lsst.analysis.tools.actions.vector.calcShapeSize import CalcShapeSize
+from lsst.analysis.tools.actions.vector.selectors import (
+    CoaddPlotFlagSelector,
+    FlagSelector,
+    GalaxySelector,
+    SkyObjectSelector,
+    SnSelector,
+    StarSelector,
+    VectorSelector,
+)
+from lsst.analysis.tools.actions.vector.vectorActions import (
+    DownselectVector,
+    ExtinctionCorrectedMagDiff,
+    FractionalDifference,
+    LoadVector,
+    MagColumnNanoJansky,
+    MagDiff,
+)
+
+
+class TestScalarActions(unittest.TestCase):
+    """ "Test ScalarActions"""
+
+    def setUp(self):
+        x = np.arange(100, dtype=float)
+        x[31] = np.nan
+        x[41] = np.nan
+        x[59] = np.nan
+        y = x**2
+        self.data = {"r_y": x, "i_y": y}
+        self.mask = np.zeros(100, dtype=bool)
+        self.mask[1:50] = 1
+
+    def _testScalarActionAlmostEqual(self, cls, truth, maskedTruth):
+        action = cls(vectorKey="{band}_y")
+        schema = [col for col, colType in action.getInputSchema()]
+        self.assertEqual(schema, ["{band}_y"])
+        result = action(self.data, band="i")
+        self.assertAlmostEqual(result, truth)
+        result = action(self.data, mask=self.mask, band="i")
+        self.assertAlmostEqual(result, maskedTruth)
+
+    def testMedianAction(self):
+        self._testScalarActionAlmostEqual(MedianAction, 2500, 576)
+
+    def testMeanAction(self):
+        self._testScalarActionAlmostEqual(MeanAction, 3321.9278350515465, 803.8936170212766)
+
+    def testStdevAction(self):
+        self._testScalarActionAlmostEqual(StdevAction, 2984.5855649976297, 733.5989754407842)
+
+    def testSigmaMadAction(self):
+        self._testScalarActionAlmostEqual(SigmaMadAction, 3278.033505115886, 759.0923358748682)
+
+    def testCountAction(self):
+        self._testScalarActionAlmostEqual(CountAction, 97, 47)
+
+    def testApproxFloorAction(self):
+        self._testScalarActionAlmostEqual(ApproxFloor, 9216.0, 2352.5)
+
+
+class TestVectorActions(unittest.TestCase):
+    """Test VectorActions"""
+
+    def setUp(self):
+        r = np.arange(5) + 1
+        i = r**2
+        rFlag = np.ones(5)
+        rFlag[1] = 0
+        iFlag = np.ones(5)
+        iFlag[3] = 0
+        data = {
+            "r_vector": r,
+            "i_vector": i,
+            "r_flag": rFlag,
+            "i_flag": iFlag,
+            "g_vector": 3 * r,
+            "E(B-V)": r[::-1],
+            "r_ixx": r**2,
+            "r_iyy": r[::-1] ** 2,
+            "r_ixy": r * r[::-1],
+            "g_iixx": r**2,
+            "g_iiyy": r[::-1] ** 2,
+            "g_iixy": r * r[::-1],
+        }
+
+        self.data = pd.DataFrame.from_dict(data)
```

Comment:
Why did you turn this into a DataFrame? This should be fine as is as a dictionary...

Reply:
I just wanted to test with a DataFrame. Before I did, there was at least one test that worked with a Dict but not a DataFrame, so it seemed worth including.